apps/projection/src/app/ui/card/card.component.html

```diff
@@ -1,29 +1,22 @@
-<div class="border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3" [class]="customClass">
-  <img
-    *ngIf="type === CardType.TEACHER"
-    src="assets/img/teacher.png"
-    width="200px"
-  />
-  <img
-    *ngIf="type === CardType.STUDENT"
-    src="assets/img/student.webp"
-    width="200px"
-  />
+<div
+  class="border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3"
+  [class]="customClass">
+  <ng-content select="[student]"></ng-content>
+  <ng-content select="[teacher]"></ng-content>
+  <ng-content select="[city]"></ng-content>
 
   <section>
     <app-list-item
       *ngFor="let item of list"
-      [name]="item.firstname"
+      [name]="item.firstname ? item.firstname : item.name"
       [id]="item.id"
-      [type]="type"
-    >
+      [type]="type">
     </app-list-item>
```

Comment:
having condition statement is not good for generic component. Everytime you will add a new card, you will have to check if the condition is still true or not. The goal is to create a very customizable list. 
You should go and learn about ngTemplateOutlet.